packages/react/src/components/Tooltip/index.tsx

```diff
@@ -143,35 +138,35 @@ export default function Tooltip({
     return () => {
       targetElement.removeEventListener('keyup', handleEscape);
     };
-  }, [showProp]);
+  }, [showTooltip]);
 
+  // Handle hover and focus events for the targetElement
   useEffect(() => {
-    if (typeof showProp !== undefined) {
-      targetElement?.addEventListener('mouseenter', handleTriggerMouseEnter);
-      targetElement?.addEventListener('mouseleave', handleTriggerMouseLeave);
-      targetElement?.addEventListener('focusin', show);
-      targetElement?.addEventListener('focusout', hide);
-    }
+    targetElement?.addEventListener('mouseenter', show);
+    targetElement?.addEventListener('mouseleave', hide);
+    targetElement?.addEventListener('focusin', show);
+    targetElement?.addEventListener('focusout', hide);
+
     return () => {
       targetElement?.removeEventListener('mouseenter', show);
       targetElement?.removeEventListener('mouseleave', hide);
       targetElement?.removeEventListener('focusin', show);
       targetElement?.removeEventListener('focusout', hide);
     };
-  }, [targetElement, showProp]);
+  }, [targetElement, show, hide]);
```

Comment:
We need to update our listeners when there are changes to the `show` and `hide` callbacks.